### ISSUE TITLE
control caching per source

### ISSUE THREAD

**Issue by 0x53A** [fsprojects/Paket#2598]
What I want:

```
source .\MyLocalNugets cache: no
source https://api.nuget.org/v3/index.json
cache: .\MyNugetCache versions:all
```

Our buildservers are offline (or rather, are not guaranteed to have internet connection), so we include all our packages in the repository. We use perforce, so the file size is not an issue (perforce has _other_ issues).

A few of our packages are custom built and not available on nuget.

Currently our paket.dependencies looks like this:

```
source .\MyNugetCache
source https://api.nuget.org/v3/index.json
cache .\MyNugetCache versions:all
```

We just manually dump the custom built nugets in the cache directory.

This works, but has one disadvantage: The lockfile becomes undiffable.

On the first resolve, the packages are not yet in the cache, so they are resolved from nuget.org.
On subsequent resolves, they are taken from the cache (because it is the first source).

This causes them to flip between the source groups in the lockfile.

At the moment, if I split it up like above, the packages would be duplicated (once in MyLocalNugets, then copied into MyNugetCache).

**Comment by matthid** [comment]
Can you add another source only for the packages which are not available on nuget? (ie duplicate those)

And use the cache folder for cache only. this way it should work...

**Comment by 0x53A** [comment]
Yes, that's what I want:

```
source .\MyLocalNugets # source for custom nugets
source https://api.nuget.org/v3/index.json # online source
cache: .\MyNugetCache versions:all
```

but wouldn't this cause all packages from MyLocalNugets  to be copied into MyNugetCache?

**Comment by matthid** [comment]
Yes they would be duplicates. That's basically what a cache is ;)
Though I can see why that is a bit unfortunate in this situation.
